### PR TITLE
Disallow point rendering with undefined point size

### LINKED ISSUE #2818: Require gl_PointSize for points rendering or define a default value
When a vertex shader doesn't set `gl_PointSize` value (while drawing using `gl.POINTS` mode), the behavior is undefined by OpenGL ES:

GLSL ES 1.0 (section 7.1):
> `mediump float gl_PointSize; // may be written to`
> If any of these variables are not written to, their values are undefined. 

GLSL ES 3.0 (section 7.1):
> If `gl_PointSize` is not written to, its value is undefined in subsequent pipe stages.

Chrome and Firefox show different behavior: Chrome doesn't render anything while Firefox seems to default to `1.0`. In spirit of #2780, the WebGL spec should be adjusted.

### PR DESCRIPTION
**Pull request by lexaknyazev**
Fixes #2818.

Should the CTS update be included in this PR before merging?

### CODE REVIEW

File: specs/latest/1.0/index.html [hunk 1]
```diff
@@ -4455,6 +4459,17 @@ <h3>Wide point primitive clipping</h3>
...
 </div>
 
+<h3>Undefined Point Size</h3>
+<p>
+    Draw calls with <code>POINTS</code> primitives are rejected when the bound program does not set
+    <code>gl_PointSize</code>.
+</p>
+<div class="note rationale">
+    <p>
+        Point rendering with undefined point size is undefined in GLES and may be unsupported in other APIs.
```
Comment (kenrussell) [inline specs/latest/1.0/index.html:4469]: From concall: please mention something more precise about why this doesn't work.

File: specs/latest/1.0/index.html [hunk 2]
```diff
@@ -3286,6 +3286,8 @@ <h4><a name="WRITING_TO_THE_DRAWING_BUFFER">Writing to the drawing buffer</a></h
             <dd>
             If <em>first</em> is negative, an <code>INVALID_VALUE</code> error will be generated.
             If the CURRENT_PROGRAM is null, an <code>INVALID_OPERATION</code> error will be generated.
+            If the <em>mode</em> is <code>POINTS</code> and the current program does not set
+            <code>gl_PointSize</code> value, an <code>INVALID_OPERATION</code> error will be generated.
```
Comment (kenrussell) [inline specs/latest/1.0/index.html:3290]: ...set the gl_PointSize variable? Here and below.

**Review by kenrussell — COMMENTED** [review]
Yes, it would be ideal to update the conformance tests at the same time.

**Review by kenrussell — CHANGES_REQUESTED** [review]
@lexaknyazev is it feasible for you to make the requested changes?

File: specs/latest/1.0/index.html (hunk 1 above)
Comment (lexaknyazev) [reply specs/latest/1.0/index.html:4469]: After looking into this again, it seems there are two underlying issues there:
- D3D11 does not support point rendering with point size greater than `1.0`, so ANGLE generates a geometry shader to handle that.
- The presence of `gl_PointSize` in the original ESSL source non-trivially affects ANGLE's shader translator (e.g., how varyings are packed), so simply prepending all shaders with `gl_PointSize = 1.0;` may cause unexpected issues. Those may be caused by unknown bugs in the translator.

So this issue looks like an ANGLE limitation, which may remain indefinitely.

What would be the appropriate language for the spec here?

**Comment by greggman** [comment]
Just so this doesn't get lost, I'm not sure this should happen. See https://github.com/KhronosGroup/WebGL/pull/2822#issuecomment-984949138

**Comment by lexaknyazev** [comment]
Closed in favor of #3370.

**Pull request closed without merging**